### ISSUE TITLE
Aurelia Framework - How to manually create a singleton instance shared across view models

### ISSUE THREAD

**Issue by andy1547** [aurelia/framework#815]
Sorry for posting here but I was unable to get a response to this question on Glitter.

I'm using container.registerInstance('MyApiClient', new MyApiClient('url-not-known-until-runtime')); in main.configure to register a global singleton which my view models can share. I'm able to inject this in my viewmodels but its constructed with no parameters, not the one registered. I've tried using both @autoinject and @inject(MyApiClient) annotations.

In short, I'm simply after a way to create a singleton instance of an object at boot time and share it across my components, how do I do this? FYI I'm using TypeScript.

**Comment by Aniel** [comment]
See http://aurelia.io/hub.html#/doc/article/aurelia/dependency-injection/latest/dependency-injection-basics/5 

`container.registerSingleton(key: any, fn?: Function): void`
 - This method allows you to register a class as a singleton. This is the default, as discussed above, so there's rarely a reason to call this method. It is provided in the API for completeness. When calling, provide the key that will be used to look up the singleton and the class which should be used. It's common for the key and class to be the same. If they are the same, then only the key needs to be provided.

**Comment by andy1547** [comment]
Had to supply the Type instead of the Type string i.e.
aurelia.container.registerSingleton(Test, () => new Test('hello'));
